api/types/state.go

```diff
@@ -21,20 +21,23 @@ type ConfigurationState string
 
 // Reasons a resource is or is not ready.
 const (
-	Authorizing                          ConfigurationState = "Authorizing"
-	ProviderNotFound                     ConfigurationState = "ProviderNotFound"
-	ProviderNotReady                     ConfigurationState = "ProviderNotReady"
-	ConfigurationStaticCheckFailed       ConfigurationState = "ConfigurationSpecNotValid"
-	Available                            ConfigurationState = "Available"
-	ConfigurationProvisioningAndChecking ConfigurationState = "ProvisioningAndChecking"
-	ConfigurationDestroying              ConfigurationState = "Destroying"
-	ConfigurationApplyFailed             ConfigurationState = "ApplyFailed"
-	ConfigurationDestroyFailed           ConfigurationState = "DestroyFailed"
-	ConfigurationReloading               ConfigurationState = "ConfigurationReloading"
-	GeneratingOutputs                    ConfigurationState = "GeneratingTerraformOutputs"
-	InvalidRegion                        ConfigurationState = "InvalidRegion"
-	TerraformInitError                   ConfigurationState = "TerraformInitError"
-	InvalidGitCredentialsSecretReference ConfigurationState = "InvalidGitCredentialsSecretReference"
+	Authorizing                                         ConfigurationState = "Authorizing"
+	ProviderNotFound                                    ConfigurationState = "ProviderNotFound"
+	ProviderNotReady                                    ConfigurationState = "ProviderNotReady"
+	ConfigurationStaticCheckFailed                      ConfigurationState = "ConfigurationSpecNotValid"
+	Available                                           ConfigurationState = "Available"
+	ConfigurationProvisioningAndChecking                ConfigurationState = "ProvisioningAndChecking"
+	ConfigurationDestroying                             ConfigurationState = "Destroying"
+	ConfigurationApplyFailed                            ConfigurationState = "ApplyFailed"
+	ConfigurationDestroyFailed                          ConfigurationState = "DestroyFailed"
+	ConfigurationReloading                              ConfigurationState = "ConfigurationReloading"
+	GeneratingOutputs                                   ConfigurationState = "GeneratingTerraformOutputs"
+	InvalidRegion                                       ConfigurationState = "InvalidRegion"
+	TerraformInitError                                  ConfigurationState = "TerraformInitError"
+	InvalidGitCredentialsSecretReference                ConfigurationState = "InvalidGitCredentialsSecretReference"
+	InvalidTerraformCredentialsSecretReference          ConfigurationState = "InvalidTerraformCredentialsSecretReference"
+	InvalidTerraformRegistryConfigMapReference          ConfigurationState = "InvalidTerraformRegistryConfigMapReference"
```

Comment:
If I understand correctly, all constant, variables, and functions named "TerraformRegistryConfigMapReference" refer to `.terraformrc` file stored in a ConfigMap). But we shouldn't refer it as simply `Registry` because this file contains more configuration [more than just registry](https://developer.hashicorp.com/terraform/cli/config/config-file#cli-configuration-file-terraformrc-or-terraform-rc) than registry information. It would be more appropriate to refer to it as "TerraformRC".

Reply:
fixed as per review comments.